cucumber/authenticators_azure/features/authn_status_azure.feature

```diff
@@ -0,0 +1,151 @@
+
+    - !grant
+      role: !group conjur/authn-azure/prod/operators
+      member: !user alice
+    """
+    And I am the super-user
+    And I successfully set Azure variables
+    And I login as "alice"
+    When I GET "/authn-azure/prod/cucumber/status"
+    Then the HTTP response status code is 200
+    And the authenticator status check succeeds
+
+  Scenario: A non-responsive Azure AD provider returns a 500 response
+    Given a policy:
+    """
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+      - !webservice
+
+      - !webservice
+        id: status
+        annotations:
+          description: Status service to verify the authenticator is configured correctly
+
+      - !variable
+        id: provider-uri
+
+      - !group users
+
+      - !permit
+        role: !group users
+        privilege: [ read, authenticate ]
+        resource: !webservice
+
+      - !group
+        id: operators
+        annotations:
+          description: Group of users who can check the status of the authn-azure/prod authenticator
+
+      - !permit
+        role: !group operators
+        privilege: [ read ]
+        resource: !webservice status
+
+    - !user alice
+
+    - !grant
+      role: !group conjur/authn-azure/prod/users
+      member: !user alice
+
+    - !grant
+      role: !group conjur/authn-azure/prod/operators
+      member: !user alice
+    """
+    And I am the super-user
+    And I successfully set Azure provider-uri variable to value "https://not-responsive.com"
+    And I login as "alice"
+    When I GET "/authn-azure/prod/cucumber/status"
+    Then the HTTP response status code is 500
+    And the authenticator status check fails with error "ProviderDiscoveryFailed: CONJ00011E"
+
+  Scenario: provider-uri variable is missing and a 500 error response is returned
+    Given a policy:
+     """
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+      - !webservice
+
+      - !webservice
+        id: status
+        annotations:
+          description: Status service to verify the authenticator is configured correctly
+
+      - !group users
+
+      - !permit
+        role: !group users
+        privilege: [ read, authenticate ]
+        resource: !webservice
+
+      - !group
+        id: operators
+        annotations:
+          description: Group of users who can check the status of the authn-azure/prod authenticator
+
+      - !permit
+        role: !group operators
+        privilege: [ read ]
+        resource: !webservice status
+
+    - !user alice
+
+    - !grant
+      role: !group conjur/authn-azure/prod/users
+      member: !user alice
+
+    - !grant
+      role: !group conjur/authn-azure/prod/operators
+      member: !user alice
+    """
+    And I am the super-user
+    And I login as "alice"
+    When I GET "/authn-azure/prod/cucumber/status"
+    Then the HTTP response status code is 500
+    And the authenticator status check fails with error "RequiredResourceMissing: CONJ00036E"
```

Comment:
Where are these error codes documented?

Reply:
in error.rb

Reply:
Is there a ticket to get these into our docs?  They would really help troubleshooting.

Reply:
Nothing like that is planned as far as I know. @alexkalish @InbalZilberman can you prioritize this?